src/policy/contract_policy.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_POLICY_CONTRACT_POLICY_H
+#define BITCOIN_POLICY_CONTRACT_POLICY_H
+
+#include <consensus/amount.h>
+#include <policy/packages.h>
+#include <policy/policy.h>
+#include <primitives/transaction.h>
+#include <txmempool.h>
+
+#include <string>
+
+// This module enforces rules for transactions with nVersion=3 ("V3 transactions") which are
+// intended for use in contracting protocols.
+
+/** Maximum virtual size of a tx which spends from a V3 transaction, in vB. */
+static constexpr unsigned int V3_CHILD_MAX_SIZE{1000};
+/** Maximum number of transactions including a tx and its descendants. */
+static constexpr unsigned int V3_DESCENDANT_LIMIT{2};
+
+// Define additional values in case we want V3 ancestor limits to diverge from default ancestor limits.
+/** Maximum number of transactions including a tx and all its mempool ancestors. */
+static constexpr unsigned int V3_ANCESTOR_LIMIT{DEFAULT_ANCESTOR_LIMIT};
+/** Maximum total virtual size of transactions, in KvB, including a tx and all its mempool ancestors. */
+static constexpr unsigned int V3_ANCESTOR_SIZE_LIMIT_KVB{DEFAULT_ANCESTOR_SIZE_LIMIT_KVB};
+
+/** Every transaction that spends an unconfirmed V3 transaction must also have V3. Check this rule
+ * for ptx and a package that may contain its unconfirmed ancestors. It's fine if ptx is one of the
+ * transactions in the package, and it's fine if none of the transactions are ancestors of ptx.
```

Comment:
It looks like part of this comment should be on the CheckV3Inheritance overload below, the one that takes a `ptx` as argument?

Reply:
Ah yes :+1: thanks!